org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/servlet/HttpProxy.java

```diff
@@ -670,4 +722,46 @@ public void cancelled() {
       }
     };
   }
+
+  /**
+   * Cookie store which uses {@link #m_defaultCookieStore} with all operations run for the specified {@link ISession}.
+   * May be used if async threads access the cookie store where {@link ISession#CURRENT} is not set correctly.
+   */
+  public class SpecificSessionCookieStore implements CookieStore {
+
+    private ISession m_session;
+
+    public SpecificSessionCookieStore(ISession session) {
+      m_session = session;
+    }
+
+    @Override
+    public void addCookie(Cookie cookie) {
+      runWithSession(() -> getDefaultCookieStore().addCookie(cookie));
```

Comment:
Did you change this from `m_defaultCookieStore` to `getDefaultCookieStore()` intentionally? The method createHttpContext() still directly accesses the member. Maybe this should both use the same pattern?

Reply:
Using `getDefaultCookieStore()` everywhere (adds the possibility to change behavior in sub-classes).